### PR TITLE
Mail library improvements

### PR DESCRIPTION
**Pull request by melnikovi**

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/architecture?pullRequest=144) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/architecture?pullRequest=144) before we can accept your contribution.<br/>

**Comment by navarr** [comment]
@woutersamaey If I'm reading the RFC correctly, I think the From can have multiple addresses.

```
from            =       "From:" mailbox-list CRLF
mailbox-list    =       (mailbox *("," mailbox)) / obs-mbox-list
```

[Definition of mailbox-list](https://tools.ietf.org/html/rfc2822#section-3.4)
[Definition of from](https://tools.ietf.org/html/rfc2822#section-3.6.2)

**Comment by woutersamaey** [comment]
@navarr OK. Not sure how what would translate in practise for Magento, but I don't object :-)

File: design-documents/mail-library-improvements.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+1. Improving `\Magento\Framework\Mail\MailMessageInterface`
+    Below is the comparison with `Zend\Mail\Message`.
+    
+    - `isValid` - does’t need to be part of interface
+    + `setEncoding`/`getEncoding` - can pass in constructor
+    - `setHeaders`/`getHeaders`
+    + `setFrom`
+    - `getFrom`
+    - `setTo` - there is addCc, but not convenient
```
Comment (joni-jones) [inline design-documents/mail-library-improvements.md]: As `\Magento\Framework\Mail\MailMessageInterface` extends `\Magento\Framework\Mail\MessageInterface`, which already has `addCc`, does it make sense to introduce a new one method?

**Comment by joni-jones** [comment]
@melnikovi could you describe how a composition of interfaces will look like in the code if logic should send HTML email with attachments?

File: design-documents/mail-library-improvements.md (hunk 1 above)
Comment (melnikovi) [reply design-documents/mail-library-improvements.md]: I would introduce it, since `addCc` doesn't allow to replace.

**Comment by melnikovi** [comment]
@joni-jones as interfaces are wrappers around `Zend\Mail` client code will look something like this https://github.com/zendframework/zend-mail/blob/master/docs/book/message/attachments.md.

**Pull request merged**